### PR TITLE
Add proxy handler for DELETE method

### PR DESCRIPTION
**Pull request by v2kk**
## Description

- Add support for DELETE method on proxyserver

## Additional context and related issues
- Trino client REST API [use DELETE to nextUri terminates a running query](https://trino.io/docs/current/develop/client-protocol.html#http-methods)

## Release notes
( x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by v2kk** [comment]
@willmostly yes, in my case, query submitted from airflow result in `trino.exceptions.HttpError: error 405`, the cancel HTTP request does not handled by Trino Gateway

**Comment by willmostly** [comment]
@v2kk can you add a test, for example https://github.com/trinodb/trino-gateway/compare/main...willmostly:trino-gateway:will/test-delete

**Review by willmostly — CHANGES_REQUESTED** [review]
The code looks good - please add a test and then this can be merged

**Review by willmostly — APPROVED** [review]
LGTM

**Comment by mosabua** [comment]
@v2kk did you submit the CLA via email already? We have to wait until it is received and processed before we can merge. cc @martint

**Review by mosabua — APPROVED** [review]
I think this is good. The one thing I think we should consider is adding some javadoc in there that explains why a DELETE request is ever used .. its required for terminating running queries.

**Review by oneonestar — APPROVED** [review]

**Comment by oneonestar** [comment]
Do we also need to handle other methods like PUT/HEAD?

**Comment by v2kk** [comment]
@mosabua I just submitted the CLA via email this morning

@oneonestar look like we only need to support 3 methods: GET, POST, DELETE. Trino python client library did it: https://github.com/trinodb/trino-python-client/blob/505989a59e93f6319e8b6be89a16f81034a06404/trino/client.py#L549

**Review by Chaho12 — APPROVED** [review]
LGTM.

**Comment by Chaho12** [comment]
same for [go client](https://github.com/trinodb/trino-go-client/blob/master/trino/trino.go), uses GET POST DELETE

**Comment by mosabua** [comment]
Could you process the CLA of @v2kk please @martint .. we would like to get this merged and cut the 10 release this week if possible.

**Comment by mosabua** [comment]
@cla-bot check

**Comment by mosabua** [comment]
@cla-bot check

**Pull request merged**
